### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by justinsalamon**
Overhaul docs to complete transition from mirdata

### CODE REVIEW

**Comment by justinsalamon** [comment]
@magdalenafuentes here's the PR, we can both push stuff here

**Comment by magdalenafuentes** [comment]
#35 related

**Comment by magdalenafuentes** [comment]
@justinsalamon I started doing quite some changes here. You can take a look at the rendered version in `docs/readthedocs.org:soundata — Read the Docs build succeeded!` in the checks. 

I'm simplifying things a bit, merging stuff, adding stuff, changing style, etc. I will be working on it in the following days. Let me know if you have comments!

**Comment by magdalenafuentes** [comment]
Closes #35, closes #11, closes #1, fixes #10, closes #11, closes #5, closes #72 and fixes #27

@justinsalamon @pzinemanas @mrocamora @genisplaja @nkundiushuti  docs are ready to review!

If any of you can take a look in the following days, this is the only thing left for release! (I removed anything that's not functionality of docs, since loaders are not _necessary_ for the release)

File: docs/index.rst [hunk 1]
```diff
@@ -1,79 +1,107 @@
...
 
-.. toctree::
-   :caption: API documentation
-   :maxdepth: 1
+**TBA**
+
+If you refer to soundata's design principles, motivation etc., please cite the following
+`paper <https://arxiv.org/abs/2109.12690>`_  [#]_:
+
+.. [#] Fuentes, M., Salamon, J., Zinemanas, P., Rocamora, M., Paja, G., Román, I.R., Miron, M., Serra, X. and Bello, J.P., 2021. Soundata: A Python library for reproducible use of audio datasets. arXiv e-prints, pp.arXiv-2109.
```
Comment (genisplaja) [inline docs/index.rst]: Plaja, G :)

File: docs/source/tutorial.rst [hunk 2]
```diff
@@ -33,165 +30,154 @@ Print a list of all available dataset loaders by calling:
...
 
-.. code-block:: python
+    .. code-block:: python
 
-    cante100.download(partial_download=['spectrogram', 'melody', 'metadata'])
+        dataset = soundata.initialize('tau2019uas')
+        dataset.download(partial_download=['development.audio.1', 'development.audio.2'])  # download only two remotes
 
 Validating a dataset
 ^^^^^^^^^^^^^^^^^^^^
```
Comment (genisplaja) [inline docs/source/tutorial.rst:158]: In addition to `Downloading a dataset into the default folder`, `Downloading a dataset into a specified folder` and `Partially downloading a dataset`, should we add here, before the validating section, an additional section that goes like: `Downloading a multipart dataset`? Basically to let them know that this feature is available and how it works.

File: docs/index.rst (hunk 1 above)
Comment (magdalenafuentes) [reply docs/index.rst]: Damn, I hope I didn't propagate this much more :o

File: docs/source/tutorial.rst (hunk 2 above)
Comment (magdalenafuentes) [reply docs/source/tutorial.rst:158]: Yep, you're right

File: docs/source/contributing_examples/example.py [hunk 3]
```diff
@@ -8,24 +8,21 @@
     1. Describe annotations included in the dataset
     2. Indicate the size of the datasets (e.g. number files and duration, hours)
     3. Mention the origin of the dataset (e.g. creator, institution)
-    4. Describe the type of music included in the dataset
-    5. Indicate any relevant papers related to the dataset
-    6. Include a description about how the data can be accessed and the license it uses (if applicable)
+    4. Indicate any relevant papers related to the dataset
+    5. Include a description about how the data can be accessed and the license it uses (if applicable)
```
Comment (nkundiushuti) [inline docs/source/contributing_examples/example.py:12]: do you consider dataset version?

File: docs/source/contributing.rst [hunk 4]
```diff
@@ -578,42 +578,40 @@ We use the following libraries for loading data from files:
...
+it will populate a clip's hidden ``_clip_metadata`` field, which can be accessed from
+the clip class.
 
 For example, if ``_metadata`` returns a dictionary of the form:
 
 .. code-block:: python
 
     {
-        'track1': {
+        'clip1': {
```
Comment (nkundiushuti) [inline docs/source/contributing.rst:604]: maybe put a soundata specific example replacing artists and genre with 'microphone-type', 'recording-date'?

File: docs/source/contributing.rst [hunk 5]
```diff
@@ -623,17 +621,51 @@ the ``_track metadata`` for ``track_id=track2`` will be:
...
+
+There should be no additional logic in a clip property/cached property, and instead all logic
+should be done in the load method. We separate these because the clip properties are only usable
 when data is available locally - when data is remote, the load methods are used instead.
 
 Missing Data
 ------------
-If a Track has a property, for example a type of annotation, that is present for some tracks and not others,
-the property should be set to `None` when it isn't available.
+If a Clip has a property, for example a type of annotation, that is present for some clips and not others,
```
Comment (nkundiushuti) [inline docs/source/contributing.rst]: I would rephrase this to make it more clear: Clip properties that are available for some clips and not for others should be set to `None` when whey are not available.

File: docs/source/faq.rst [hunk 6]
```diff
@@ -1,66 +1,77 @@
...
-----------------------
-soundata = mir + data. MIR is an acronym for Music Information Retrieval, and the library was built for working with data.
 
+.. admonition:: Does soundata support data which lives off-disk?
+    :class: dropdown
 
-If I find a mistake in an annotation, should I fix it in the loader?
---------------------------------------------------------------------
-No. All datasets have "mistakes", and we do not want to create another version of each dataset ourselves. The loaders should load the data as released. After that, it's up to the user what they want to do with it.
+    Yes. While the simple useage of soundata assumes that data lives on-disk, it can be used for off-disk data as well.
```
Comment (nkundiushuti) [inline docs/source/faq.rst:73]: this is not clear for me. what do you mean fix it in the loader? you change the annotation.csv to correct the annotation and you update the Index in the loader on your machine right? because if you do any change to the loader's index in soundata the dataset would not validate anymore

File: docs/source/overview.rst [hunk 7]
```diff
@@ -10,7 +10,8 @@ Overview
```
Comment (nkundiushuti) [inline docs/source/overview.rst:10]: btw the link for the notebooks doesn't work. maybe add a Coming soon...

File: docs/source/overview.rst [hunk 8]
```diff
@@ -10,7 +10,8 @@ Overview
```
Comment (nkundiushuti) [inline docs/source/overview.rst:11]: spotted a typo: soundata provdes ``Annotation`` classes of various kinds which provide a standard interface to different annotation formats such as tags and sound events.

Comment (nkundiushuti) [reply docs/source/overview.rst:11]: another typo: When available, we provide extensive and easy-to-access ``metadata`` to facilitate clip metadata-specific analysis. 
``metadata`` is available as attroibutes at the ``clip`` level, e.g. ``clip.fold``.

Comment (nkundiushuti) [reply docs/source/overview.rst:11]: `clipgroups` are used when a dataset comprises **of** clipgroups

File: docs/source/tutorial.rst [hunk 9]
```diff
@@ -33,165 +30,202 @@ Print a list of all available dataset loaders by calling:
...
+            "development.audio.3": download_utils.RemoteFileMetadata(
+                filename="TAU-urban-acoustic-scenes-2019-development.audio.3.zip",
+                url="https://zenodo.org/record/2589280/files/TAU-urban-acoustic-scenes-2019-development.audio.3.zip?download=1",
+                checksum="c7214a07211f10f3250290d05e72c37e",
             ),
-        }
+            ....
 
-An partial download example for ``cante100`` dataset could be:
+    An partial download example for ``tau2019uas`` dataset could be:
```
Comment (nkundiushuti) [inline docs/source/tutorial.rst]: A partial?

File: docs/source/tutorial.rst [hunk 10]
```diff
@@ -33,165 +30,202 @@ Print a list of all available dataset loaders by calling:
...
+    example_tags = example_clip.tags
+    print(example_tags)
+    >>>> Tags(confidence, labels, labels_unit)
+    print(example_tags.labels)
+    >>>> ['children_playing']
 
 
-Alternatively, we don't need to load the whole dataset to get a single track.
+
+You can also load a single clip without loading all clips int the dataset:
```
Comment (nkundiushuti) [inline docs/source/tutorial.rst]: in the dataset

**Review by nkundiushuti — APPROVED** [review]

File: docs/source/contributing.rst (hunk 4 above)
Comment (magdalenafuentes) [reply docs/source/contributing.rst:604]: good catch!

File: docs/source/faq.rst (hunk 6 above)
Comment (magdalenafuentes) [reply docs/source/faq.rst:73]: what this means is that we do not want to change the data ourselves, since we are not the authors of most datasets we won't be able to change the source. This is to avoid that a new user that gets the dataset from source (with the mistake) would have a different version that the fixed one and the dataset would not validate. The safest is to open an issue and discuss per case

File: docs/source/overview.rst (hunk 7 above)
Comment (magdalenafuentes) [reply docs/source/overview.rst:10]: good catch, will remove altogether for now

**Comment by magdalenafuentes** [comment]
@nkundiushuti thanks for all the comments! I addressed them and I'm merging this
